pkg/resmgr/control/cpu/cpu.go

```diff
@@ -271,28 +265,26 @@ func (ctl *cpuctl) configure(cfg *cfgapi.Config) error {
 
 	// Configure the system
 	for class, cpus := range assignments {
-		if _, ok := ctl.classes[class]; ok {
-			// Re-configure cpus (sysfs) according to new class parameters
-			if err := ctl.enforceCpufreq(class, cpus.SortedMembers()...); err != nil {
-				log.Error("cpufreq enforcement on re-configure failed: %v", err)
-			}
-			if err := ctl.enforceCpufreqGovernor(class, cpus.SortedMembers()...); err != nil {
-				log.Error("cpufreq enforcement on re-configure failed: %v", err)
-			}
-		} else {
+		if _, ok := ctl.classes[class]; !ok {
 			// TODO: what should we really do with classes that do not exist in
 			// the configuration anymore? Now we remember the CPUs assigned to
 			// them. A further config update might re-introduce the class in
 			// which case the CPUs will be reconfigured.
-			log.Warn("class %q with cpus %v missing from the configuration", class, cpus)
+			return fmt.Errorf("non-existent cpu class %q", class)
```

Comment:
The reason for changing this from a warning to `ErrorF` is that, while the original code doesn't raise an error immediately, it would eventually fail when calling `enforceCpufreq` or `enforceCpufreqGovernor` if the class isn't valid. By restructuring the code this way, we fail early.

Reply:
aThe logic was originally:
```
for each class and associated cpus:
    if there is cpu configuration for this class, apply it (and if applying fails, log an error but **don't tell it to the caller**),
    otherwise log a warning that cpu configuration for a class is missing.
    Then continue with the next class in the next round of the loop.
```

This change proposes logic:
```
for each class and associated cpus:
    if there is no cpu configuration for this class, bail out and **return an error to the caller**, leave remaining classes and cpus unhandled (yet they would have configurations), and forget about possible uncore configurations after the for loop.
    apply the configuration (and if it fails, log an error but don't tell it to the caller)
```

And if you look at the caller of configure() (see Start()), you can see that what used to be a warning will now prevent starting the controller. And if you go up in the call stack (StartStopControllers(), startControllers()....) that now missing CPU properties for single class prevents starting the whole resource manager (policy).

So I think this change would fail early, and do it in a very fatal way.

I'd prefer the original, quite forgiving handling of missing cpu class information. Maybe we can make these issues more visible to the user through policy status at some point. However, those changes would be topics for other PRs.